editor/src/core/shared/github/helpers.ts

```diff
@@ -846,14 +872,21 @@ export async function refreshGithubData(
   previousCommitSha: string | null,
   originCommitSha: string | null,
   operationContext: GithubOperationContext,
-): Promise<void> {
+): Promise<Array<Promise<Array<EditorAction>>>> {
```

Comment:
This type feels wrong, a Promise of an array of Promises is functionally equivalent to an array of Promises.

I think this should be refactored slightly so that the `await` is reworked into a `.then` as this function is effectively blocking with the `await` and then turning the result into a resolved promise.

Reply:
nice catch!